### PR TITLE
VULCAN-88 VULCAN-65 Recover Liferay APIs / Support for language

### PR DESCRIPTION
**Pull request by ahdezma**
Until the name of the project is decided recover all Liferay APIs.

Remove `Group` API and use `Site` API instead (thanks to @victorlaerte).

### CODE REVIEW

File: vulcan-jaxrs-json/src/main/java/com/liferay/vulcan/jaxrs/json/internal/writer/WriterHelper.java [hunk 1]
```diff
@@ -377,6 +378,48 @@ public String getAbsoluteURL(
...
+		).ifPresent(
+			fieldFunctions -> {
+				for (String field : fieldFunctions.keySet()) {
+					if (fieldsPredicate.test(field)) {
+						BiFunction<T, Language, String> fieldFunction =
+							fieldFunctions.get(field);
+
+						String data = fieldFunction.apply(model, language);
+
+						if ((data != null) && !data.isEmpty()) {
```
Comment (nhpatt) [inline vulcan-jaxrs-json/src/main/java/com/liferay/vulcan/jaxrs/json/internal/writer/WriterHelper.java]: parenthesis are not needed here

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+ * Provides all the necessary information to expose <a
+ * href="http://schema.org/DigitalDocument">DigitalDocument</a> resource through
+ * a web API. <p> The resources are mapped from the internal {@link DLFileEntry}
+ * model.
+ *
+ * @author Javier Gamarra
+ * @review
+ */
+@Component(immediate = true, service = CollectionResource.class)
+public class DigitalDocumentScopedCollectionResource
```
Comment (nhpatt) [inline vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java]: Shouldn't this be MediaObjectCollectionResource?

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.ServerErrorException;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Provides all the necessary information to expose <a
+ * href="http://schema.org/DigitalDocument">DigitalDocument</a> resource through
```
Comment (nhpatt) [inline vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java]: MediaObject?

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+		).addString(
+			"text", DLFileEntry::getDescription
+		).addType(
+			"MediaObject"
+		).build();
+	}
+
+	@Override
+	public String getName() {
+		return "digitalDocuments";
```
Comment (nhpatt) [inline vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java]: :P

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/SiteCollectionResource.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+ * href="http://schema.org/WebSite">Web Site</a> resources through a web API.
+ *
+ * The resources are mapped from the internal {@link Site} model.
+ *
+ * @author Victor Oliveira
+ * @author Alejandro Hernández
+ * @review
+ */
+@Component(immediate = true)
+public class SiteCollectionResource
```
Comment (nhpatt) [inline vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/SiteCollectionResource.java]: Shouldn't be this WebSite?

**Comment by ahdezma** [comment]
Totally agree, it would've been better to make two PR. Haste is the Devil's work.

File: vulcan-jaxrs-json/src/main/java/com/liferay/vulcan/jaxrs/json/internal/writer/WriterHelper.java [hunk 6]
```diff
@@ -377,6 +378,48 @@ public String getAbsoluteURL(
...
+			modelClass, fields);
+
+		Optional<Representor<T, Identifier>> optional =
+			_collectionResourceManager.getRepresentorOptional(modelClass);
+
+		optional.map(
+			Representor::getLocalizedStringFunctions
+		).ifPresent(
+			fieldFunctions -> {
+				for (String field : fieldFunctions.keySet()) {
```
[the remark this answers is not in the document]
Comment (ahdezma) [reply vulcan-jaxrs-json/src/main/java/com/liferay/vulcan/jaxrs/json/internal/writer/WriterHelper.java]: Thanks :)

https://github.com/liferay/com-liferay-vulcan/pull/79/commits/45788aa159ad48bfc313d1a6b789198315e0bdb4

File: vulcan-jaxrs-json/src/main/java/com/liferay/vulcan/jaxrs/json/internal/writer/WriterHelper.java (hunk 1 above)
Comment (ahdezma) [reply vulcan-jaxrs-json/src/main/java/com/liferay/vulcan/jaxrs/json/internal/writer/WriterHelper.java]: SourceFormatter forces it

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java (hunk 3 above)
Comment (ahdezma) [reply vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java]: https://github.com/liferay/com-liferay-vulcan/pull/79/commits/79dd537043af050955d2068611eab4d85e5acfe3

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java (hunk 2 above)
Comment (ahdezma) [reply vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java]: https://github.com/liferay/com-liferay-vulcan/pull/79/commits/79dd537043af050955d2068611eab4d85e5acfe3

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java (hunk 4 above)
Comment (ahdezma) [reply vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/DigitalDocumentScopedCollectionResource.java]: ;)
https://github.com/liferay/com-liferay-vulcan/pull/79/commits/79dd537043af050955d2068611eab4d85e5acfe3

File: vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/SiteCollectionResource.java (hunk 5 above)
Comment (ahdezma) [reply vulcan-sample-liferay-portal/src/main/java/com/liferay/vulcan/sample/liferay/portal/internal/resource/SiteCollectionResource.java]: https://github.com/liferay/com-liferay-vulcan/pull/79/commits/79dd537043af050955d2068611eab4d85e5acfe3

**Comment by ahdezma** [comment]
ci:retest

**Comment by nhpatt** [comment]
👍

**Pull request merged**